src/Auth.ts

```diff
@@ -188,17 +252,19 @@ export class Auth {
       }
     }
 
-    let getTokenPromise: ((resource: string, logger: Logger, debug: boolean, fetchNew: boolean) => Promise<AccessToken | null>) | undefined;
+    let getTokenPromise: ((resource: string, logger: Logger, debug: boolean, fetchNew: boolean) => Promise<AuthenticationResult | null>) | undefined;
```

Comment:
For retrieving an accessToken, why do we need identityId?

Reply:
In reference to what line is this question? Or is it a general question? 

The idea is that we need to select the correct msal account when refreshing tokens. The identityId is the linking pin between our identity objects and the msal token store.

Reply:
We changed the return type from AccessToken to AuthenticationResult which contains identityId.

identityId is already available on the service, right? So why does it also need to be a part of the returned access token?

Reply:
The `identityId` property on the `AuthenticationResult` is used to fill the `identityId` property on the `service` object.

Reply:
Isn't that exposed to us separately by MSAL or do we have to get it as part of the token retrieval?

Reply:
When retrieving a token from MSAL, we may get a response with the account (delegated) or without an account (app-only).

In both scenario's we need to get the identityId. The ideal moment to get it would be right there after acquiring the token and retrieving the AuthenticationResult from MSAL. Which is what I'm doing here. I don't think there's a way to get the identityId from MSAL in other ways. You have to get the account or the token...